### DISCUSSION TITLE
Buttons on an external I2C or SPI GPIO extender

### DISCUSSION THREAD

**Discussion by SnowballsChance** [HomeSpan/HomeSpan#108] [Q&A] [answered]
I am using several [RX480E Receivers](https://qiachip.com/products/2x-learning-code-receiver-module-for-rf-433mhz-rx-480-e-remote-control-arduino-chip-28131mm-pcb) connected via an [MCP23017 GPIO Extender](https://ww1.microchip.com/downloads/en/DeviceDoc/20001952C.pdf). Currently I have it working via MQTT to Homebridge. 

How can I co-opt HomeSpan's  `new SpanButton(powerPin);` for this?

I imagine I might need to override some of what HomeSpan offers... How? 

Thanks for having a look.

**Comment by HomeSpan** [comment, marked as answer]
I don't think you need to use SpanButton() since it is designed primarily to debounce physical buttons.  If you are using a GPIO extender to read remote button presses (or similar), you'll need to interface directly with the GPIO extender using either I2C or SPI (depending on the version of the extender you are using).  The standard Arduino libraries, `Wire.h` and `SPI.h`, both work well within the HomeSpan environment.  You'll want to implement your code within the `loop()` method of a *HomeSpan Service.*  A good example of using I2C in this fashion can be found in [TemperatureSensorI2C](https://github.com/HomeSpan/TempSensorI2C).

**Comment by SnowballsChance** [comment]
Thank you! Really appreciate the new example! 
Will update with progress in a couple of days.

**Comment by SnowballsChance** [comment]
oops I forgot to come back and say how it went... 

It all works well. My code if  can be found [here](https://github.com/SnowballsChance/HS-Buttons).

Thanks for your help.
